mlflow/java/spark/pom.xml

```diff
@@ -1,13 +1,17 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
-  <artifactId>mlflow-spark</artifactId>
+  <artifactId>mlflow-spark_${scala.compat.version}</artifactId>
   <version>2.8.2-SNAPSHOT</version>
   <name>${project.artifactId}</name>
   <properties>
     <maven.compiler.source>1.8</maven.compiler.source>
     <maven.compiler.target>1.8</maven.compiler.target>
     <encoding>UTF-8</encoding>
-    <scala.version>2.12.6</scala.version>
+    <!-- Set default scala version to 2.12 and override it with 2.13 in one profile for DBR support.
+      We still need 2.12 jar for pyspark support as it still uses scala 2.12 in the latest version (3.5)
+      and raises an exception due to imcomptibility if mlflow-spark jar is built with 2.13.
+      This is temporary fix until (Py)Spark 4 release that will use scala 2.13 by default. -->
+    <scala.version>2.12.18</scala.version>
     <scala.compat.version>2.12</scala.compat.version>
```

Comment:
A reminder on follow-up task: we need to update MLflow release pipeline, to update stage of java releasing, so that it can release both scala 2.12 and 2.13 version